packages/lib/file-api-driver-amazon-s3.js

```diff
@@ -175,6 +175,12 @@ class FileApiDriverAmazonS3 {
 	}
 
 	metadataToStats_(mds) {
+		// aws-sdk-js-v3 can rerturn undefined instead of an empty array when there is
+		// no metadata in some cases.
+		//
+		// Thus, we handle the !mds case.
+		if (!mds) return [];
```

Comment:
Probably this fix is more proper than modify the response object, but I don't really see how it can fix the issue?

Or was it because of the strict equality with undefined, when response.Contents was actually "null" or something similar?

Reply:
Not included in the diff, because it was unchanged, we have
```ts

		while (response.IsTruncated) {
			response = await this.s3ListObjects(prefixPath, response.NextContinuationToken);
			output = output.concat(this.metadataToStats_(response.Contents, prefixPath));
		}
```
which doesn't include the check for `response.Contents === undefined`.

Reply:
Oh right, I missed that so that's good then. The check on metadataToStats_ makes sense